Src/Common/CorrelationIdLookupHelper.cs

```diff
@@ -245,26 +233,75 @@ private Task<string> FetchAppIdFromService(string instrumentationKey)
                 {
                     SdkInternalOperationsMonitor.Enter();
 
+                    string result = null;
                     Uri appIdEndpoint = this.GetAppIdEndPointUri(instrumentationKey);
-
                     WebRequest request = WebRequest.Create(appIdEndpoint);
                     request.Method = "GET";
 
                     using (HttpWebResponse response = (HttpWebResponse)request.GetResponse())
                     {
-                        using (StreamReader reader = new StreamReader(response.GetResponseStream()))
+                        if (response.StatusCode == HttpStatusCode.OK)
+                        {
+                            using (StreamReader reader = new StreamReader(response.GetResponseStream()))
+                            {
+                                result = reader.ReadToEnd();
+                            }
+                        }
+                        else
                         {
-                            return reader.ReadToEnd();
+                            this.RegisterFetchFailure(instrumentationKey, response.StatusCode);
                         }
                     }
+
+                    return result;
                 }
                 finally
                 {
                     SdkInternalOperationsMonitor.Exit();
                 }
             });
         }
+#else
+        /// <summary>
+        /// Retrieves the app id given the instrumentation key.
+        /// </summary>
+        /// <param name="instrumentationKey">Instrumentation key for which app id is to be retrieved.</param>
+        /// <returns>App id.</returns>
+        private async Task<string> FetchAppIdFromService(string instrumentationKey)
+        {
+            try
+            {
+                SdkInternalOperationsMonitor.Enter();
+
+                string result = null;
+                Uri appIdEndpoint = this.GetAppIdEndPointUri(instrumentationKey);
+                WebRequest request = WebRequest.Create(appIdEndpoint);
+                request.Method = "GET";
+
+                using (HttpWebResponse response = (HttpWebResponse)await request.GetResponseAsync().ConfigureAwait(false))
+                {
+                    if (response.StatusCode == HttpStatusCode.OK)
```

Comment:
wouldn't `GetResponseAsync` throw an exception when the response is not `200`?

Reply:
I can't find documentation that states GetResponse will throw an exception for non 200 responses.
But I think we're covered in this scenario as well because the encapsulating code is catching and logging all exceptions.

Do you want us to handle exceptions differently?